### DISCUSSION TITLE
Tips on migrating from Keycloak?

### DISCUSSION THREAD

**Discussion by TarekAS** [ory/kratos#2484] [Q&A] [answered]
Hi. We are currently using Keycloak as an Identity Provider/AuthN server only (the most basic use case, i.e. no other features used). We are looking to adopt the Ory stack and start by migrating the users from Keycloak to Kratos.

Has anyone done this before? If so, any tips or guidance would be greatly appreciated.

**Comment by aeneasr** [comment, marked as answer]
A good place to start is the „importing identities“ guide:

https://www.ory.sh/docs/guides/manage-identities/import-identity

We do not have a dedicated guide for migrating though. It’s a frequent topic though, maybe @vinckr  can help out to write and write a guide together?

**Reply by TarekAS** [reply]
A dedicated guide would be much appreciated!

**Reply by vinckr** [reply]
Hello @TarekAS 
Happy to collaborate on a document for this. Tbh I have never used Keycloak, so if we could do the migration together that would be helpful. 
Maybe you can make some notes on the procedure and I help out when you run into roadblocks. Also happy to jump on a [quick call](https://meetings-eu1.hubspot.com/vincent-kraus) to discuss it first - or feel free to DM me in our [community slack](https://slack.ory.sh/)!

**Reply by TarekAS** [reply]
When we get to this task I'll be happy to collaborate. Will get in touch, thanks!
